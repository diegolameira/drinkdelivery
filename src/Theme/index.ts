import styledTS from 'styled-components-ts';

interface ThemeInterface {
  background: string;
  foreground: string;
  primary: string;
  danger: string;
}

export class Theme implements ThemeInterface {
  background = '#fff';
  foreground = '#000';
  primary = '#FCD635';
  danger = '#DB0000';
}

interface stylInterface {
  theme: ThemeInterface;
}

export const styl = Component => styledTS<stylInterface>(Component);
