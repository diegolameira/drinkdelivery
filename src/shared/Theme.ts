import styledTS from 'styled-components-ts';

export class Theme implements ThemeInterface {
  background = '#fff';
  foreground = '#000';
  primary = '#FCD635';
  danger = '#DB0000';
}

export const styl = Component => styledTS<stylInterface>(Component);
