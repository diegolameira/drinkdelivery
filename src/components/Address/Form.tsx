import React, { Component } from 'react';

import { throttle } from '@/shared/Helpers';
import Input from '@/components/Input';
import { List } from '@/components/List';

import { Place, Suggestion } from './Interfaces';
import { getPlaceSuggestionsByTerm, getPlaceDetails } from './Helpers';

import { Wrapper } from './Form.styles';

interface Props {
  googleKey: string;
  label?: string;
  placeholder?: string;
  onUpdate: (place: Place) => void;
}

interface State {
  suggestions: Suggestion[];
}

export class AddressForm extends Component<Props, State> {
  lazySearch: (term?: string) => void;

  constructor(props: Props) {
    super(props);
    this.lazySearch = throttle(this.search, 400);
  }

  search = (term: string) => {
    if (!term || term.length <= 2) return;
    getPlaceSuggestionsByTerm(this.props.googleKey, term).then(suggestions =>
      this.setState({
        suggestions
      })
    );
  };

  onSelect = (placeId: string) => {
    this.setState({
      suggestions: []
    });
    getPlaceDetails(this.props.googleKey, placeId)
      .then(selected => {
        this.setState({
          value: selected.title
        });
        return selected;
      })
      .then(this.props.onUpdate);
  };

  render() {
    const { suggestions, value } = this.state || {};
    const items =
      suggestions &&
      suggestions.length > 0 &&
      suggestions.map(({ placeId, ...rest }) => ({
        ...rest,
        // replace id with placeId for <List />
        id: placeId
      }));
    return (
      <Wrapper>
        <Input
          {...this.props}
          value={value}
          onChangeText={text => this.lazySearch(text)}
        />
        {items && <List items={items} onSelect={this.onSelect} />}
      </Wrapper>
    );
  }
}
