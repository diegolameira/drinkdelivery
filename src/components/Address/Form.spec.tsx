/* eslint-env jest */

import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Theme } from '@/Theme';

import { AddressForm } from './Form';

Enzyme.configure({ adapter: new Adapter() });

describe(`AddressForm`, () => {
  const theme = new Theme();
  const component = renderer.create(<AddressForm theme={theme} />);
  const wrapper = shallow(<AddressForm theme={theme} />);
  let tree: ReactTestRendererJSON = component.toJSON();

  beforeEach(() => {});
  it('should render as expected', () => {
    const props = {
      label: 'Label',
      placeholder: 'placeholder'
    };
    expect(tree).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();

    tree.props = props;
    wrapper.setProps(props);

    expect(tree).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();
  });
});
