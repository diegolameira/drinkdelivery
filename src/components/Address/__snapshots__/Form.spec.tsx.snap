// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressForm should render as expected 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "left": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "right": 0,
        "top": 0,
        "zIndex": 9999,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 10,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 15,
            "fontWeight": "300",
            "height": 0,
            "lineHeight": 15,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "opacity": 0.5,
          },
          undefined,
        ]
      }
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="default"
      onChangeText={[Function]}
      selectTextOnFocus={true}
      style={
        Array [
          Object {
            "borderColor": "rgba(0, 0, 0, 0.2)",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#000",
            "fontSize": 20,
            "minWidth": 70,
            "paddingBottom": 18,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 18,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`AddressForm should render as expected 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddressForm
    theme={
      Theme {
        "background": "#fff",
        "danger": "#DB0000",
        "foreground": "#000",
        "primary": "#FCD635",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          onChangeText={[Function]}
          selectTextOnFocus={true}
          theme={
            Theme {
              "background": "#fff",
              "danger": "#DB0000",
              "foreground": "#000",
              "primary": "#FCD635",
            }
          }
        />,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChangeText": [Function],
          "selectTextOnFocus": true,
          "theme": Theme {
            "background": "#fff",
            "danger": "#DB0000",
            "foreground": "#000",
            "primary": "#FCD635",
          },
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            onChangeText={[Function]}
            selectTextOnFocus={true}
            theme={
              Theme {
                "background": "#fff",
                "danger": "#DB0000",
                "foreground": "#000",
                "primary": "#FCD635",
              }
            }
          />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChangeText": [Function],
            "selectTextOnFocus": true,
            "theme": Theme {
              "background": "#fff",
              "danger": "#DB0000",
              "foreground": "#000",
              "primary": "#FCD635",
            },
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`AddressForm should render as expected 3`] = `
<View
  label="Label"
  placeholder="placeholder"
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 10,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 15,
            "fontWeight": "300",
            "height": 0,
            "lineHeight": 15,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "opacity": 0.5,
          },
          undefined,
        ]
      }
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="default"
      onChangeText={[Function]}
      selectTextOnFocus={true}
      style={
        Array [
          Object {
            "borderColor": "rgba(0, 0, 0, 0.2)",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#000",
            "fontSize": 20,
            "minWidth": 70,
            "paddingBottom": 18,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 18,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`AddressForm should render as expected 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddressForm
    label="Label"
    placeholder="placeholder"
    theme={
      Theme {
        "background": "#fff",
        "danger": "#DB0000",
        "foreground": "#000",
        "primary": "#FCD635",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          label="Label"
          onChangeText={[Function]}
          placeholder="placeholder"
          selectTextOnFocus={true}
          theme={
            Theme {
              "background": "#fff",
              "danger": "#DB0000",
              "foreground": "#000",
              "primary": "#FCD635",
            }
          }
        />,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Label",
          "onChangeText": [Function],
          "placeholder": "placeholder",
          "selectTextOnFocus": true,
          "theme": Theme {
            "background": "#fff",
            "danger": "#DB0000",
            "foreground": "#000",
            "primary": "#FCD635",
          },
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            label="Label"
            onChangeText={[Function]}
            placeholder="placeholder"
            selectTextOnFocus={true}
            theme={
              Theme {
                "background": "#fff",
                "danger": "#DB0000",
                "foreground": "#000",
                "primary": "#FCD635",
              }
            }
          />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Label",
            "onChangeText": [Function],
            "placeholder": "placeholder",
            "selectTextOnFocus": true,
            "theme": Theme {
              "background": "#fff",
              "danger": "#DB0000",
              "foreground": "#000",
              "primary": "#FCD635",
            },
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
